### DISCUSSION TITLE
Garage door controller (MSG100 & MSG200) - toggle trigger enabled state?

### DISCUSSION THREAD

**Discussion by pengee** [krahabb/meross_lan#475] [Ideas]
In the Meross app there are configuration settings for automatic closing via overtime (length of time open) and at a fixed time. I use the overtime setting because in most cases, the garage door doesn't need to be open for more than say 30 minutes. However, it would be great to be able to connect this to a camera's detection state of people in the garage. So in the case that I'm actively working in the garage for more than 30 minutes, the door will not try to close and the state will be re-enabled when leaving.

In the Appliance.System.All response under system/digest/ there is a triggerx entry with listings for 'overtimeclose' and a timerx entry with listings that I would thing are for the fixed time settings.

Would it be possible to enable control of these settings in Home Assistant, or could you help with what a manual payload would look like to toggle them?

**Comment by krahabb** [comment]
@Arandall in https://github.com/arandall/meross/blob/main/doc/protocol.md has some knowledge about the digest.triggerx settings and controls.

It need to be tested though but I guess you could use `Appliance.Control.TriggerX` namespace (method `SET`) to configure your 'triggerx' entry.
Since the digest.triggerx contains a collection of triggers you should be able to individually setup/update every single entry by sending something in the line..
```json
{
"triggerx": {
    "id": "yourtriggerxid",
    "enable": 0
  }
}
```
I'm not sure this would just update the configuration of the existent trigger or overwrite everything (in the latter case you'd have to send the full trigger configuration as it is now on the device and just change the "enabled" field to the intended value...

You could use the `meross_lan.request` service to send the aforementioned payload and whatever query you would need. I've never tested in real but since the service provides the device response I think you will be able to use it in HA scripts or automations.


In my opinion, it would be maybe better to completely drop usage of the Meross embedded triggers and rely on an HA automation to fullfill the same behavior. This way, of course, your Meross device by itself would not be able to behave automatically: it would always require HA to be up and running to manage this...

**Comment by pengee** [comment]
Thank you, this was very helpful! Apologies for not finding the docs on this, that was indeed my mistake.

I agree that having HA control this behavior is preferable, but that has some failure points (HA state, WIFI). For now, having a toggle button on the HA dash is a good starter.

Changing this setting does work but does require most of the config, not just the id / state. The only odd thing seems to be that the device and cloud state fall out of alignment if you update the device locally.
